### ISSUE TITLE
V2: authenticate() function doesn't handle string credentialIds

### ISSUE THREAD

**Issue by hjaber** [passwordless-id/webauthn#65]
The `authenticate()` function in the client library assumes that the `allowCredentials` array contains objects with an `id` property. However, when passing an array of credential ID strings directly, the function throws a TypeError.

```ts
   import { client } from '@passwordless-id/webauthn'
   // Call the `authenticate()` function with an array of credential ID strings:
   const authentication = await client.authenticate({
     challenge: "server-generated-challenge",
     allowCredentials: ["credentialId1", "credentialId2"],
     timeout: 60000
   })
```
Results in `TypeError: "Cannot read properties of undefined (reading 'replaceAll')"`

src/client.ts `authenticate()` diff
```diff ts
- id: utils.parseBase64url(id),
+ id: utils.parseBase64url(cred.id)
```
```ts
interface CredentialDescriptor {
    id: Base64URLString;
    transports?: AuthenticatorTransport[];
}
```

**Comment by dagnelies** [comment]
should be fixed in 2.0.2

**Comment by hjaber** [comment]
Fixed in 2.0.2, thank you!
